conjure-java-core/src/integrationInput/java/com/palantir/product/NameCollisionServiceEndpoints.java

```diff
@@ -104,4 +110,103 @@ public HttpHandler handler() {
             return this;
         }
     }
+
+    private static final class NoContextEndpoint implements HttpHandler, Endpoint {
+        private final UndertowRuntime runtime;
+
+        private final UndertowNameCollisionService delegate;
+
+        private final Deserializer<String> deserializer;
+
+        NoContextEndpoint(UndertowRuntime runtime, UndertowNameCollisionService delegate) {
+            this.runtime = runtime;
+            this.delegate = delegate;
+            this.deserializer = runtime.bodySerDe().deserializer(new TypeMarker<String>() {});
+        }
+
+        @Override
+        public void handleRequest(HttpServerExchange exchange) throws IOException {
+            String requestContext = deserializer.deserialize(exchange);
+            delegate.noContext(requestContext);
+            exchange.setStatusCode(StatusCodes.NO_CONTENT);
+        }
+
+        @Override
+        public HttpString method() {
+            return Methods.POST;
+        }
+
+        @Override
+        public String template() {
+            return "/no/context";
+        }
+
+        @Override
+        public String serviceName() {
+            return "NameCollisionService";
+        }
+
+        @Override
+        public String name() {
+            return "noContext";
+        }
+
+        @Override
+        public HttpHandler handler() {
+            return this;
+        }
+    }
+
+    private static final class ContextEndpoint implements HttpHandler, Endpoint {
+        private static final ImmutableSet<String> TAGS = ImmutableSet.of("server-request-context");
+
+        private final UndertowRuntime runtime;
+
+        private final UndertowNameCollisionService delegate;
+
+        private final Deserializer<String> deserializer;
+
+        ContextEndpoint(UndertowRuntime runtime, UndertowNameCollisionService delegate) {
+            this.runtime = runtime;
+            this.delegate = delegate;
+            this.deserializer = runtime.bodySerDe().deserializer(new TypeMarker<String>() {});
+        }
+
+        @Override
+        public Set<String> tags() {
+            return TAGS;
```

Comment:
Out of curiosity, why are tags available at runtime?

Reply:
They're incredibly useful at runtime to opt into custom HttpHandler wrappers, we used them with a couple products to apply a wrapper that opts into less desirable behavior to match old Jersey bugs without forcing it upon new endpoints.

The SLO and limits features we've discussed would likely leverage tags initially as well -- they're a great staging ground for experimental features without adding new blocks to the conjure lang, IR, and reference implementations.